src/transformers/file_utils.py

```diff
@@ -1926,6 +1926,11 @@ def push_to_hub(
                 Specify this in case you want to push to an existing repository in the hub. If unspecified, a new
                 repository will be created in your namespace (unless you specify an :obj:`organization`) with
                 :obj:`repo_name`.
+            working_directory (:obj:`str`, `optional`):
+                Folder where the repository has already been cloned (if it exists). This is useful if you are pushing
+                to an existing repo, to avoid cloning it again, but it might replace some files in that folder if you
+                are not up to date (like would happen in a :obj:`git pull`), so you should only use it if you don't
+                have uncommited files inside.
```

Comment:
Should this also be used when the repository has not been cloned? Should it be mentioned that this will also push all files in the `working_directory`?